core/domain/topic_domain.py

```diff
@@ -250,6 +341,26 @@ def validate(self):
                 'Expected description to be a string, received %s'
                 % self.description)
 
+        if not isinstance(self.subtopics, list):
+            raise utils.ValidationError(
+                'Expected subtopics to be a list, received %s'
+                % self.subtopics)
+
+        if not isinstance(self.schema_version, int):
+            raise utils.ValidationError(
+                'Expected schema version to be an integer, received %s'
+                % self.schema_version)
+
+        for subtopic in self.subtopics:
+            if not isinstance(subtopic, Subtopic):
+                raise utils.ValidationError(
+                    'Expected each subtopic to be a Subtopic object, '
+                    'received %s' % subtopic)
+            subtopic.validate()
+            for skill_id in subtopic.skill_ids:
+                if skill_id in self.uncategorized_skill_ids:
+                    self.uncategorized_skill_ids.remove(skill_id)
```

Comment:
Validation should not mutate. Instead check that every skill in every subtopic + the uncategorized skill IDs list is distinct.

Reply:
Yeah, I thought of that, but even if they some overlap, there is no need for an error, right? As we can just remove the conflicting id from uncategorized_skill_ids?

Reply:
But validation should not mutate, so this operation should happen during the "move" operation itself. And validation just checks that the results of these operations result in a correct and internally-consistent thing.

Reply:
Ok, will shift this to another function.

Reply:
Done.